nilearn/surface/surface.py

```diff
@@ -1014,3 +1014,229 @@ def check_surface(surface):
+
+    dtype : numpy dtype-like or None, optional
+        The dtype that should be used for the returned sparse matrix.
+
+    Returns
+    -------
+    matrix : scipy.sparse.csr_matrix
+        A sparse matrix representing the edge relationships in `surface`.
+
+    """
+    from scipy.sparse import csr_matrix
+    surface = load_surf_mesh(surface)
+    # This is a bit of a hack to quickly find a unique set of all edges.
+    n = surface.coordinates.shape[0]
+    edges = np.vstack([surface.faces[:,[0,1]],
+                       surface.faces[:,[0,2]],
+                       surface.faces[:,[1,2]]])
+    edges = edges.astype(np.int64)
+    bigcol = edges[:,0] > edges[:,1]
+    lilcol = ~bigcol
+    edges = np.concatenate([edges[bigcol,0] + edges[bigcol,1]*n,
+                            edges[lilcol,1] + edges[lilcol,0]*n])
+    edges = np.unique(edges)
+    (u,v) = (edges // n, edges % n)
+    # Calculate distances between pairs. We use this as a weighting to make sure that
+    # smoothing takes into account the distance between each vertex neighbor
+    if values == 'len' or values == 'invlen':
+        coords = surface.coordinates
+        edge_lens = np.sqrt(np.sum((coords[u,:] - coords[v,:])**2, axis=1))
+        if dtype is None:
+            dtype = edge_lens.dtype
+        else:
+            edge_lens = edge_lens.astype(dtype)
+        if values == 'invlen':
+            edge_lens = 1 / edge_lens
+    elif values == 'ones':
+        if dtype is None:
+            edge_lens = np.ones_like(edges)
+        else:
+            edge_lens = np.ones(edges.shape, dtype=dtype)
+    else:
+        raise ValueError(f"unrecognized values argument: {values}")
+    # We can now make a sparse matrix.
+    ee = np.concatenate([edge_lens, edge_lens])
+    uv = np.concatenate([u, v])
+    vu = np.concatenate([v, u])
+    return csr_matrix((ee, (uv, vu)), shape=(n,n))
+
+
+def _compute_vertex_neighborhoods(surface):
+    """For each vertex, compute the neighborhood.
+    The neighborhood is  defined as all the vertices that are connected by a
+    face.
+    
+    Parameters
+    ----------
+    surface : Surface-like
+        The surface whose vertex neighborhoods are to be computed.
+    
+    Returns
+    -------
+    neighbors : list
+        A list of all the vertices that are connected to each vertex
+    """
+    from scipy.sparse import find
+    matrix = _compute_adjacency_matrix(surface)
+    return [find(row)[1] for row in matrix]
+
+       
+def smooth_surface_data(surface, surf_data,
+                        iterations=1,
+                        distance_weights=False,
+                        vertex_weights=None,
+                        return_vertex_weights=False,
+                        center_surround_knob=0,
+                        match='sum'):
+    """Smooth values along the surface.
+    
+    Parameters
+    ----------
+    surface : Surface-like
+        The surface on which the data `surf_data` are to be smoothed.
```

Comment:
Can we be more specific here? And is this the same input as the `surface` parameter under e.g. `load_surface` and `check_surface` ? If so, it might be useful to add a standard docstring entry in `_utils.docs.py`